sql/api/query.go

```diff
@@ -398,17 +400,45 @@ func (p *QueryParameterDateTimeSec) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
-// QueryParameterDateRange ...
-type QueryParameterDateRange struct {
+type DateTimeRange struct {
+	Start string `json:"start"`
+	End   string `json:"end"`
+}
+
+type QueryParameterRangeBase struct {
 	QueryParameter
 
-	Value string `json:"value"`
+	Value       any            `json:"value"`
+	StringValue string         `json:"-"`
+	RangeValue  *DateTimeRange `json:"-"`
+}
+
+func (p *QueryParameterRangeBase) toParameterObject() {
+	if p.RangeValue != nil {
+		p.Value = p.RangeValue
+	} else {
+		p.Value = p.StringValue
+	}
+}
+
+func (p *QueryParameterRangeBase) decodeQueryParameter() {
+	if v, ok := p.Value.(map[string]any); ok {
+		p.RangeValue = &DateTimeRange{Start: v["start"].(string), End: v["end"].(string)}
+	} else {
+		p.StringValue = fmt.Sprintf("%v", p.Value)
+	}
```

Comment:
You could set value to nil from here instead of right after this call.

Reply:
oh, yes - it was an artifact where I did first it as a separate function, not function on the type